src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/ReconciliationService.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.hmcts.reform.blobrouter.reconciliation.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.reform.blobrouter.data.reconciliation.statements.SupplierStatementRepository;
+import uk.gov.hmcts.reform.blobrouter.data.reconciliation.statements.model.NewEnvelopeSupplierStatement;
+import uk.gov.hmcts.reform.blobrouter.exceptions.InvalidSupplierStatementException;
+import uk.gov.hmcts.reform.blobrouter.reconciliation.model.in.SupplierStatement;
+
+import java.sql.SQLException;
+import java.time.LocalDate;
+import java.util.UUID;
+
+@Service
+public class ReconciliationService {
+
+    private final SupplierStatementRepository repository;
+    private final ObjectMapper objectMapper;
+
+    public ReconciliationService(SupplierStatementRepository repository, ObjectMapper objectMapper) {
+        this.repository = repository;
+        this.objectMapper = objectMapper;
+    }
+
+    @Transactional
```

Comment:
i wonder whether it needs to be here or on actual `save` method :thinking:

Reply:
the repository method doesn't have it. It is added to service methods in the existing classes

Reply:
I think service should determine the transaction boundary rather than repository